### PR TITLE
Don't do vpdupdate for updatenode

### PR DESCRIPTION
**Pull request by zet809**
### The modification include

Check lsvpd and do vpdupdate when deploying.

### The UT result
Before fix:
```
Jan 16 04:58:21 mid21tor24cn02 xcat.updatenode: INFO  =============updatenode starting====================
Jan 16 04:58:21 mid21tor24cn02 lsvpd[22498]: INFO  libvpd: Unable to stat vpd.db file.
Jan 16 04:58:21 mid21tor24cn02 lsvpd[22498]: INFO  libvpd: Running vpdupdate to update the default db.
Jan 16 04:58:21 mid21tor24cn02 lsvpd[22500]: NOTICE  vpdupdate: Constructing full devices database
...
Jan 16 04:58:58 mid21tor24cn02 xcat.updatenode: INFO  =============xcatdsklspost update VPD====================
Jan 16 04:58:58 mid21tor24cn02 xcat.updatenode: INFO  trying to download postscripts...
Jan 16 04:58:58 mid21tor24cn02 xcat.updatenode: INFO  postscripts downloaded successfully
Jan 16 04:58:58 mid21tor24cn02 xcat.updatenode: INFO  trying to get mypostscript from 172.16.253.100...
Jan 16 04:58:58 mid21tor24cn02 xcat.updatenode.postscript: INFO  postscript start..: custom.ps/list_software
Jan 16 04:59:03 mid21tor24cn02 xcat.updatenode.postscript: INFO  postscript end...:custom.ps/list_software return with 1
Jan 16 04:59:03 mid21tor24cn02 xcat.updatenode: INFO  =============updatenode ending====================
```
After fix:
```
Jan 16 05:46:07 mid21tor24cn02 xcat.updatenode: INFO  =============updatenode starting====================
Jan 16 05:46:07 mid21tor24cn02 xcat.updatenode: INFO  trying to download postscripts...
Jan 16 05:46:07 mid21tor24cn02 xcat.updatenode: INFO  postscripts downloaded successfully
Jan 16 05:46:07 mid21tor24cn02 xcat.updatenode: INFO  trying to get mypostscript from 172.16.253.100...
Jan 16 05:46:07 mid21tor24cn02 xcat.updatenode.postscript: INFO  postscript start..: custom.ps/list_software
Jan 16 05:46:07 mid21tor24cn07 dhclient[126240]: INFO  DHCPDISCOVER on enP3p3s0f1 to 255.255.255.255 port 67 interval 14 (xid=0xe5a51a6)
Jan 16 05:46:12 mid21tor24cn02 xcat.updatenode.postscript: INFO  postscript end...:custom.ps/list_software return with 1
Jan 16 05:46:12 mid21tor24cn02 xcat.updatenode: INFO  =============updatenode ending====================
```

### CODE REVIEW

File: xCAT/postscripts/xcatdsklspost [hunk 1]
```diff
@@ -304,6 +284,27 @@ else
...
+{
+    if [ -f /usr/sbin/vpdupdate ]; then
+        vpdupdate
+        #logger -t xCAT -p local4.info "xcatdsklspost: updating VPD database"
+        echolog "info" "updating VPD database"
+    fi
+}
+
+if [ $NODE_DEPLOYMENT -eq 1 ] || [ "$MODE" = "4" ] || [ "$MODE" = "6" ]; then
+    echolog "info" "=============deployment starting===================="
```
Comment (robin2008) [inline xCAT/postscripts/xcatdsklspost]: It is not `deployment starting`, it is `special` log message.  Give more reasonable log there

File: xCAT/postscripts/xcatdsklspost [hunk 2]
```diff
@@ -304,6 +284,27 @@ else
...
   esac
 fi
+
+update_VPD()
+{
+    if [ -f /usr/sbin/vpdupdate ]; then
+        vpdupdate
+        #logger -t xCAT -p local4.info "xcatdsklspost: updating VPD database"
+        echolog "info" "updating VPD database"
+    fi
```
Comment (robin2008) [inline xCAT/postscripts/xcatdsklspost]: Add WARN message when `vpdupdate` is not there.

File: xCAT/postscripts/xcatdsklspost [hunk 3]
```diff
@@ -304,6 +284,27 @@ else
     3|6) MODE=$1;;
   esac
 fi
+
+update_VPD()
+{
+    if [ -f /usr/sbin/vpdupdate ]; then
+        vpdupdate
+        #logger -t xCAT -p local4.info "xcatdsklspost: updating VPD database"
+        echolog "info" "updating VPD database"
```
Comment (robin2008) [inline xCAT/postscripts/xcatdsklspost]: Here should be VPD database udpated, instead of `updating`

File: xCAT/postscripts/xcatdsklspost [hunk 4]
```diff
@@ -304,6 +284,27 @@ else
...
+    fi
+}
+
+if [ $NODE_DEPLOYMENT -eq 1 ] || [ "$MODE" = "4" ] || [ "$MODE" = "6" ]; then
+    echolog "info" "=============deployment starting===================="
+    # Run updatevpd only when necessary
+    if [ -f /usr/sbin/lsvpd ]; then
```
Comment (robin2008) [inline xCAT/postscripts/xcatdsklspost]: I would suggest to combine `-f xxx` with L297 as you only do only one task.

**Comment by zet809** [comment]
HI, @robin2008 , update according your comments. Pls review, Thx!

**Pull request merged**